### PR TITLE
Adjust container create flow

### PR DESCRIPTION
**Pull request by alpaix**
The key purpose of this PR is to adjust the container creation APIs to allow a backend via its driver generate new container ID rather than requiring a client to provide one.

As the ID is generated by the driver we need to return it to the client via the container's attach method
```ts
attach(request: IRequest): Promise<IResolvedUrl>
```

At the driver level we need to utilize the URL resolver to resolve the new document URL given its ID
```
createContainer(createNewResolvedUrl: IResolvedUrl, urlResolver: IUrlResolver, ...): Promise<IDocumentService>
```

### CODE REVIEW

File: common/lib/container-definitions/src/loader.ts [hunk 1]
```diff
@@ -156,12 +156,14 @@ export interface IContainer extends IEventProvider<IContainerEvents>, IFluidRout
...
+     * Attaches the Container to a new document created in the document storage.
+     * @param request - Request to be used to create a new container.
+     * @returns - Resolved URL associated with the new document.
      *
-     * TODO - in the case of failure options should give a retry policy. Or some continuation function
+     * TODO: - in the case of failure options should give a retry policy. Or some continuation function
      * that allows attachment to a secondary document.
      */
-    attach(request: IRequest): Promise<void>;
+    attach(request: IRequest): Promise<IResolvedUrl>;
```
Comment (anthony-murphy) [inline common/lib/container-definitions/src/loader.ts:166]: it's missing from the interface, but Container exposes it's IResolvedUrl, why can't it just be retrieved after attach?

File: common/lib/driver-definitions/src/storage.ts [hunk 2]
```diff
@@ -290,16 +290,22 @@ export interface IDocumentServiceFactory {
...
+     * @param request - Request required to create a new document.
+     * @param urlResolver - URL resolver required to resolve document URLs.
+     * @param createNewSummary - Optional. Summary used to create file.
+     * If undefined, an empty file will be created and a summary should be posted later,
+     * before connecting to ordering service.
+     * @param logger - Optional. Telemetry logger.
+     * @returns Returns the document service connected to the delta stream.
      */
     createContainer(
-        createNewSummary: ISummaryTree | undefined,
```
Comment (anthony-murphy) [inline common/lib/driver-definitions/src/storage.ts:301]: this change cannot be made. ideally it's not necessary, but if it is you need to add a new optional method and somehow support both

**Comment by tanviraumi** [comment]
@znewton Take a look on the r11s driver side change

**Comment by alpaix** [comment]
Will abandon this pr and revisit the implementation. Thanks for the feedback!

**Pull request closed without merging**